### DISCUSSION TITLE
UNPKG - Some components missing

### DISCUSSION THREAD

**Discussion by heliacer** [mit-cml/blockly-plugins#49] [Q&A] [answered]
I've tried out implementing the lexical variables using the [unpkg sourcecode](https://www.unpkg.com/@mit-app-inventor/blockly-block-lexical-variables@2.0.2/dist/index.js)
It did surprisingly work, but only partially. I have a testing demo [here](https://heliacer.github.io/blockly-lexical/)
Only the procedures category and the loops category pick up the components of lexical variables. 
the rest, including the main variables category just stays on its default.

Does anyone know why? Could it be because i'm using the new JSON layout?
Thanks for letting me know!

![preview](https://github.com/mit-cml/blockly-plugins/assets/118212730/082f049e-e601-4b41-abf7-b1851d9a2aec)

Regards, heliacer

**Comment by mark-friedman** [comment]
@heliacer Please note the following from the "Usage" section of the README file:
>Note that unlike with standard Blockly, you should not use a custom toolbox category for your variables, as this would interfere with the way that variables are declared and used with this plugin. Just create an ordinary Variables category, if you want, and place the lexical-variable-get and lexical-variable-set blocks in there.

**Comment by mark-friedman** [comment]
> Hey there, Thanks for writing back. I did read all prerequisites and am aware of that. If im correct the variables category is named as its default, so it shouldn't really interfere.

What my quoted text was trying to say is that it **does** interfere.  You shouldn't use the standard **custom** variables category.

> 
> Do i have to place the blocks manually? If thats the case then it would make sense on why the variables didn't appear.

Yes. Please take a look at toolbox definition in the [demo repo](https://github.com/mit-cml/lexical-variable-demo) for some more guidance.

**Comment by mark-friedman** [comment, marked as answer]
> Turns out its possible to add them manually to the toolbox category of "Variables". I managed to find those blocks ( excluding the mutators ) but i didn't manage to find the public variable initialising block on the unpkg source code.
> 
> ```js
>         {
>           kind: 'block',
>           type: 'lexical_variable_get'
>         },
>         {
>           kind: 'block',
>           type: 'lexical_variable_set'
>         },
>         {
>           kind: 'block',
>           type: 'local_declaration_statement'
>         },
>         {
>           kind: 'block',
>           type: 'local_declaration_expression'
>         }
> ```

All the blocks and their types should be specified in the README file for the plug-in.  For example, [here's](https://github.com/mit-cml/blockly-plugins/tree/main/block-lexical-variables#global-variable-declaration) the global variable definition block.

**Reply by heliacer** [reply]
Thanks a lot for the help!
I was a bit confused at first,
but i think i get it now. 

Have a nice day,
Regards
Heliacer

**Comment by mark-friedman** [comment]
👍
